conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosFeeOperations.scala

```diff
@@ -26,17 +26,21 @@ private[tezos] object TezosFeeOperations extends LazyLogging {
     "endorsement"
   )
 
-  /**
-    * Calculates average fees for each operation kind and stores them into a fees table.
+  /** Calculates average fees for each operation kind and stores them into a fees table.
+    * The computation will use a limited number of fees, as the result of a selection window in days.
+    * Only blocks belonging within such window in the past, relative to the calculation moment, will be considered.
+    *
     * @param numberOfFeesAveraged a limit on how many of the latest fee values will be used for averaging
-    * @param ex the needed ExecutionContext to combine multiple database operations
+    * @param selectionWindow the max number of days back from the current block timestamp to use when averaging
+    * @param ec the needed ExecutionContext to combine multiple database operations
     * @return a future result of the number of rows stored to db, if supported by the driver
     */
-  def processTezosAverageFees(numberOfFeesAveraged: Int)(implicit ex: ExecutionContext): Future[Option[Int]] = {
+  def processTezosAverageFees(selectionWindow: FiniteDuration)(implicit ec: ExecutionContext): Future[Option[Int]] = {
     logger.info("Processing latest Tezos fee data...")
+
     val computeAndStore = for {
-      fees <- DBIOAction.sequence(operationKinds.map(TezosDb.calculateAverageFees(_, numberOfFeesAveraged)))
-      dbWrites <- TezosDb.writeFees(fees.collect { case Some(fee) => fee })
+      feeStats <- TezosDb.FeesStatistics.calculateAverage(selectionWindow.toDays)
```

Comment:
Why not keep it as an arbitrary duration? It will be more general that way.

Reply:
Makes things more complex, but not too much.
Yet I wonder if that's worth it right now. It could be tuned later quite easily to afford more complex definitions.

For the moment the level of granularity in the definition amounts to a whole day.

Reply:
Makes sense.